snap/snapcraft.yaml

```diff
@@ -90,6 +90,12 @@ parts:
     stage:
     - -linux
     - -src
+    override-build: |
+      snapcraftctl build
+      [ -d ../install/linux/lib32 ] && SO_FILE_32=$(find ../install/linux/lib32/* -name "libphobos2.so.[0-9]*.[0-9]*.[0-9]*")
+      [ -d ../install/linux/lib64 ] && SO_FILE_64=$(find ../install/linux/lib64/* -name "libphobos2.so.[0-9]*.[0-9]*.[0-9]*")
+      [ -d ../install/linux/lib32 ] && [ -f $SO_FILE_32 ] && ln -sf $SO_FILE_32 ${SO_FILE_32%.*}
+      [ -d ../install/linux/lib64 ] && [ -f $SO_FILE_64 ] && ln -sf $SO_FILE_64 ${SO_FILE_64%.*}
```

Comment:
This is still not working, I'm afraid. :-(

Here is what I currently have in the existing dmd 2.085.1 snap package:

```
-rw-r--r-- 1 root root 69M Apr 10 19:41 libphobos2.a
lrwxrwxrwx 1 root root  20 Apr 10 19:41 libphobos2.so -> libphobos2.so.0.85.1
-rwxr-xr-x 1 root root 11M Apr 10 19:42 libphobos2.so.0.85.1
```

Here's what I get when I install the snap package built from this PR:

```
-rw-r--r-- 1 root root 68M Apr 17 13:58 libphobos2.a
lrwxrwxrwx 1 root root  20 Apr 17 13:58 libphobos2.so -> libphobos2.so.0.85.0
lrwxrwxrwx 1 root root  43 Apr 17 13:58 libphobos2.so.0.85 -> ../install/linux/lib64/libphobos2.so.0.85.0
-rwxr-xr-x 1 root root 11M Apr 17 13:59 libphobos2.so.0.85.0
```

Unsurprisingly, the second symlink is broken.  I suspect you need to use the `-r` option with `ln` to make sure the links are specified relative to the link location.

i.e. use

```
ln -sfr
```

rather than just `ln -sf` ;-)

Reply:
I tested `ln -sfr` locally and it results in:

```
-rw-r--r-- 1 root root 68M Apr 17 17:51 libphobos2.a
lrwxrwxrwx 1 root root  20 Apr 17 17:51 libphobos2.so -> libphobos2.so.0.85.0
lrwxrwxrwx 1 root root  20 Apr 17 17:51 libphobos2.so.0.85 -> libphobos2.so.0.85.0
-rwxr-xr-x 1 root root 11M Apr 17 17:52 libphobos2.so.0.85.0
```

... which I think is what is wanted, no ... ?